nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.integtests.dremio;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class DremioHelper {
+  private String token;
+  private String projectId;
+  private String baseUrl;
+  private String catalogName;
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  private static JsonNode parseJson(String json) throws IOException {
+    JsonFactory factory = OBJECT_MAPPER.getFactory();
+    JsonParser parser = factory.createParser(json);
+    return OBJECT_MAPPER.readTree(parser);
+  }
+
+  DremioHelper(String projectId, String token, String baseUrl, String catalogName) {
+    this.projectId = projectId;
+    this.token = token;
+    this.baseUrl = baseUrl;
+    this.catalogName = catalogName;
+  }
+
+  private String createPayload(String query) {
+    String payload =
+        format("{ \"sql\": \"%s\", \"context\": [\"%s\", \"db\"] }", query, catalogName);
```

Comment:
This won't work, if the SQL query contains for example a `"`
Better construct the payload with JsonNode and let Jackson produce the string.

Reply:
done.